backend/curriculum_tracking/models.py

```diff
@@ -1619,17 +1619,46 @@ def get_users_that_reviewed_open_prs(self):
 
         return [review.user for review in reviews]
 
-    def request_user_can_start(self):
-        """can the user start this card?
+    def request_user_can_start(self, user=None):
+        """
+        Check if current user can start this card
+
+        This function is only used in template rendering and we are using threadlocals,
+        that is why we need to avoid argument parameters unless they're for testing purposes
+        """
+        if user is None:
+            from threadlocal_middleware import get_current_user
+
+            user = get_current_user()
+
+        if user is not None:
+            request_user_is_assignee = self.assignees.first() == user
+
+            if request_user_is_assignee and self.can_start():
+                return True
+
+            has_manage_cards_permission = any(
+                [
+                    user.has_perm(Team.PERMISSION_MANAGE_CARDS, team)
+                    for team in self.assignees.first().teams()
+                ]
+            )
+
+            if has_manage_cards_permission and self.can_start():
+                return True
+
+        return False
+
+    def request_user_is_assignee(self):
```

Comment:
Not seeing where we are making use of this. We manually did this in `request_user_can_start`:

```
request_user_is_assignee = self.assignees.first() == user
```

Reply:
I'm doing it manually because there were clashes with the test environments and how we get the requesting user.

However, the actual method is still being used in the template.

I need to pass in a user and do the user checks I'm doing in the `request_user_can_start` to call it in without issues. I opted not to because that `request_user_is_assignee` is not being tested.

Maybe I'm thinking about it wrong, any suggestions are welcome.